hibernate-core/src/main/java/org/hibernate/proxy/pojo/javassist/JavassistLazyInitializer.java

```diff
@@ -57,6 +57,11 @@ public Object invoke(
 			final Method proceed,
 			final Object[] args) throws Throwable {
 		if ( this.constructed ) {
```

Comment:
Great!
Just to add that it's not just for `$$_hibernate_getEntityEntry` but for all methods added by bytecode enhancment (prefixed with `$$_hibernate_`) including the getters for the persistent attributes. All these are supposed to be called only by hibernate internally and therefore should never trigger a load.